core/boot/src/main/java/org/eclipse/dataspaceconnector/boot/system/DefaultServiceExtensionContext.java

```diff
@@ -281,7 +283,12 @@ private void addDefaultExtensions(List<ServiceExtension> loadedExtensions) {
 
         var injectFields = Arrays.stream(ext.getClass().getDeclaredFields())
                 .filter(f -> f.getAnnotation(Inject.class) != null)
-                .map(f -> new FieldInjectionPoint<>(ext, f, getFeatureValue(f.getType())));
+                .map(f -> {
+                    var isRequired = f.getAnnotation(Inject.class).required();
+                    var fip = new FieldInjectionPoint<>(ext, f, getFeatureValue(f.getType()));
+                    fip.setRequired(isRequired);
```

Comment:
I would add the `required` parameter to the constructor to avoid the set methods and leaving the `FieldInjectionPoint` object immutable